### DISCUSSION TITLE
Animated body

### DISCUSSION THREAD

**Discussion by oooreoreo** [InteractiveComputerGraphics/SPlisHSPlasH#165] [Q&A] [answered]
Hi,

In the example animated_body.py, time_step_callback() was used to make the dragon spin. I have a series of .obj files which contain the motion sequence of a rigid body. Can I  make it animate by using time_step_callback(), too? I tried to delete and add boundary models in the function, but I got confused with the classes and functions of pysph. Is it workable?  If so, how could I achieve that? Thanks!!

**Comment by janbender** [comment, marked as answer]
I think it is not the best way to remove and add boundary models since things have to be initialized in the right order. Depending on the boundary handling you use, you have different options. If you use Akinci boundary handling, you could update the positions and velocities of all boundary particles in the script. For the other methods you need a position and rotation matrix of the boundary model. For the position you could compute the center of mass in each frame and move the boundary model. Extracting the rotation is a bit more complex but this can be done using shape matching (https://cg.informatik.uni-freiburg.de/publications/2005_SIGGRAPH_deformationShapeMatching.pdf). Btw. our position-basd dynamics library contains an implementation of shape mathcing (https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/136469f03f7869666d907ea8d27872b098715f4a/PositionBasedDynamics/PositionBasedDynamics.cpp#L501).

Hope that helps!

**Reply by oooreoreo** [reply]
I tried in the first way and I can change the positions and velocities of the initialized boundary model, but I can not get positions and velocities of the model in the following frames. In my understanding, .obj files are sampled and cached as .bgeo files, and the properties of the sampled ones are useful. How could I get them in the script?

**Reply by oooreoreo** [reply]
oh, I just need to learn more about the sampling  and do it in advance, and the library can help. Thank you so much for the answer, it really helps.
